ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/BeaconState.java

```diff
@@ -147,6 +151,201 @@ public BeaconState(
     this.eth1_data_votes = eth1_data_votes;
   }
 
+  public static BeaconState fromBytes(Bytes bytes) {
+    return SSZ.decode(
+        bytes,
+        reader ->
+            new BeaconState(
+                // Misc
+                UnsignedLong.fromLongBits(reader.readUInt64()),
+                UnsignedLong.fromLongBits(reader.readUInt64()),
+                Fork.fromBytes(reader.readBytes()),
+                // Validator registry
+                (ArrayList<Validator>)
```

Comment:
I don't believe these casts after reading a bytes list are explicitly necessary. While I don't mind having them, the rest of our files don't so I'd prefer they're removed for consistency. @jrhea - thoughts?

Reply:
it isn't immediately obvious to me why the cast is needed.  it didn't look like you needed it in any of the other fromBytes() methods.  it would be nice to know the answer (i need to eat dinner so please feel free to investigate

Reply:
I do get a compilation error without them:

```
error: incompatible types: inference variable R has incompatible bounds
            new BeaconState(
            ^
    equality constraints: List<T#2>
    upper bounds: ArrayList<Validator>,Object
  where R,A,T#1,T#2 are type-variables:
    R extends Object declared in method <R,A>collect(Collector<? super T#1,A,R>)
    A extends Object declared in method <R,A>collect(Collector<? super T#1,A,R>)
    T#1 extends Object declared in interface Stream
    T#2 extends Object declared in method <T#2>toList()
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
```

I'm open to suggestions.

Reply:
Yeah, the issue is because the constructor expected ArrayLists instead of the interface type List. I'll fix it and push the fix to your branch.

Reply:
Fixed.